OpenRA.Mods.D2k/Traits/Buildings/D2kActorPreviewPlaceBuildingPreview.cs

```diff
@@ -104,7 +104,8 @@ public D2kActorPreviewPlaceBuildingPreviewPreview(WorldRenderer wr, ActorInfo ai
 				if ((c.Value & filter) == 0)
 					continue;
 
-				var isUnsafe = checkUnsafeTiles && candidateSafeTiles.Contains(c.Key) && info.UnsafeTerrainTypes.Contains(wr.World.Map.GetTerrainInfo(c.Key).Type);
+				var terrainType = wr.World.Map.Contains(c.Key) ? wr.World.Map.GetTerrainInfo(c.Key).Type : null;
+				var isUnsafe = checkUnsafeTiles && candidateSafeTiles.Contains(c.Key) && info.UnsafeTerrainTypes.Contains(terrainType);
```

Comment:
I assume this works because `UnsafeTerrainTypes` never contain `null`? Wouldn't it make more sense to just put `wr.World.Map.Contains(c.Key)` directly into the expression? Like this:
```suggestion
				var isUnsafe = checkUnsafeTiles && wr.World.Map.Contains(c.Key) && candidateSafeTiles.Contains(c.Key) && info.UnsafeTerrainTypes.Contains(terrainType);
```
Is the correct thing to display here the `unsafe` or `blocked` sequence?

Reply:
That won't compile, but okay then we get a massively too long line though.

Reply:
If you think the line is too long we could add a line break after the second `&&`. What doesn't compile? I'm also still not sure if using `isUnsafe` is correct here but will trust you on this since I haven't tested it myself (yet).

Reply:
Cells outside the map should be treated as `PlaceBuildingCellType.Invalid` anyway, so i believe that the actual value of `isUnsafe` is never going to be checked in this case. The currently pushed commit looks like I would expect the fix for this to look (but I haven't reviewed or tested it properly yet).